ckeditor.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global requirejs, define, require, window, document, location */
+
+'use strict';
+
+// This file is shared by the dev and release versions of CKEditor. It bootstraps the API.
+
+( function( root ) {
+	var CKEDITOR = root.CKEDITOR = {
+		/**
+		 * The full URL for the CKEditor installation directory.
+		 *
+		 * It is possible to manually provide the base path by setting a global variable named `CKEDITOR_BASEPATH`. This
+		 * global variable must be set **before** the editor script loading.
+		 *
+		 *		alert( CKEDITOR.basePath ); // e.g. 'http://www.example.com/ckeditor/'
+		 *
+		 * @property {String}
+		 */
+		basePath: getBasePath(),
+
+		/**
+		 * Defines an AMD module.
+		 *
+		 * See https://github.com/ckeditor/ckeditor5-design/wiki/AMD for more details about our AMD API.
+		 *
+		 * @method
+		 * @member CKEDITOR
+		 */
+		define: define,
+
+		/**
+		 * Retrieves one or more AMD modules.
+		 *
+		 * Note that the CKEditor AMD API does not download modules on demand so be sure to have their relative scripts
+		 * available in the page.
+		 *
+		 * See https://github.com/ckeditor/ckeditor5-design/wiki/AMD for more details about our AMD API.
+		 *
+		 * @method
+		 * @member CKEDITOR
+		 */
+		require: require
+	};
+
+	requirejs.config( {
+		// Modules are generally relative to the core project.
+		baseUrl: CKEDITOR.basePath + 'node_modules/ckeditor5-core/src/',
+
+		// These configurations will make no difference in the build version because the following paths will be
+		// already defined there.
+		paths: {
+			// Hide the core "ckeditor" under a different name.
+			'ckeditor-core': CKEDITOR.basePath + 'node_modules/ckeditor5-core/src/ckeditor',
+
+			// The dev version overrides for the "ckeditor" module. This is empty on release.
+			'ckeditor-dev': CKEDITOR.basePath + 'src/ckeditor-dev'
+		}
+	} );
+
+	// Define a new "ckeditor" module, which overrides the core one with the above and the dev stuff.
+	define( 'ckeditor', [ 'ckeditor-core', 'ckeditor-dev', 'utils' ], function( core, dev, utils ) {
+		utils.extend( core, root.CKEDITOR, ( dev || {} ) );
+		root.CKEDITOR = core;
+
+		return core;
+	} );
+
+	function getBasePath() {
```

Comment:
This function should be testable. We already had problem in CKEditor 4 that in order to test the base path we had to load multiple iframes.